autoload/lsp/completion.vim

```diff
@@ -590,12 +590,17 @@ def LspCompleteDone(bnr: number)
     # additional text edits.  So try to resolve the completion item now to get
     # the text edits.
     completionData = lspserver.resolveCompletion(completionData, true)
-    if completionData->get('additionalTextEdits', {})->empty()
-      return
+    if !completionData->get('additionalTextEdits', {})->empty()
+      textedit.ApplyTextEdits(bnr, completionData.additionalTextEdits)
     endif
   endif
 
-  textedit.ApplyTextEdits(bnr, completionData.additionalTextEdits)
+  if completionData->has_key('command')
+    # Some language servers (e.g. haskell-language-server) want to apply
+    # additional commands after completion.
+    lspserver.executeCommand(completionData.command)
```

Comment:
perhaps better to use `codeaction.DoCommand()` instead.

Reply:
Why? In this case it ends up just calling lspserver.executeCommand() anyway?

Reply:
yes, but just gave a self-custom entry to user if that alternative way.
or thought that not necessary? i donot know actually, for this `command`, 
but for uniform, so far there was no where directly call lspserver.execuateCommand